### ISSUE TITLE
Unexplainable Slow Chunk Loading with High Player Counts

### ISSUE THREAD

**Issue by blake-tisbury** [pl3xgaming/Purpur#729]
### Timings link

https://timings.pl3x.net/?id=d8d6587b55b746a6ad201612f633840f

### Description of issue

I and many others have been experiencing an unexplainable slow chunk loading with high player counts (greater than 50). 

The timings report, tps, and CPU usage are all very normal but the chunk loading will take seconds per chunk, in turn emulating a view distance of 1-2 chunks. With high player counts in ranges of 100, you can almost outrun the chunk rendering and reach the void. This issue seemed to have arisen around August. No amount of configuration seems to ameliorate the issue. Again, the issue isn't limited to hardware as per core CPU usage ranges from 10-30%.

### Purpur version

This server is running Purpur version git-Purpur-1418 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: 78b0bcc on ver/1.17.1)
You are running the latest version
Previous version: git-Purpur-1415 (MC: 1.17.1)

### Agreements

- [X] I am running the latest version of Purpur available from https://purpur.pl3x.net/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Purpur.

### Other

I'm attaching the spark report, a screenshot of htop running 3 servers with ~120 people, and a screenshot of the chunk generation
https://spark.lucko.me/Ah2PCmGsoz
![image](https://user-images.githubusercontent.com/61334533/138574212-ea994149-6430-4607-a651-6be5b6116500.png)
![image](https://user-images.githubusercontent.com/61334533/138574216-4a80de23-9934-4226-b340-7d1145e1ffe6.png)

**Comment by Monjiigames** [comment]
Hello.

Just to confirm I am going through the exact same issue. I have the latest purpur version avaliable and the issue is still persisting. I have a 70 av playerbase and the chunks are fine only if I kick everyone else. As OP mentioned, the server itself is not lagging, TPS, CPU usage timings all look fine.

**Comment by KaspianDev** [comment]
Paper.yml change maximum sent chunks

**Comment by Monjiigames** [comment]
> Paper.yml change maximum sent chunks

We have both tried this

**Comment by KaspianDev** [comment]
Plugins also can cause this, disable them one by one

**Comment by blake-tisbury** [comment]
> Plugins also can cause this, disable them one by one

Doesn't seem to be the factor here. Happened both on my lobby and other servers. The only overlapping plugins are Litebans, Luckperms, TAB and World Guard. I highly doubt any of these plugins would affect chunk gen and the timings don't show any apparent issues with the plugins.
I will say though, a common denominator between manji and I is the fact that we both use OVH. No idea if that is playing a role here.

**Comment by Ravi7302** [comment]
hi i am having same issue. i have disabled every plugin, but its still happening, it means its an issue with purpur fork?

**Comment by astei** [comment]
As I explained to the OP (who also talked about this issue on the Velocity Discord):
> You have compression disabled and it seems that the link between your proxy and your servers is too expensive. Either move your servers to the same box as the proxy or set the network compression threshold to 1024 on all backend servers

Indeed, I examined the timings report and confirmed compression was disabled. It seems more likely to me this was the issue rather than being a server or proxy issue that appeared out of nowhere.

**Comment by Incluuu** [comment]
@blake-tisbury maybe you have anti-xray turned on?

**Comment by Apprisco** [comment]
It's also quite possible that you are bottlenecked by the I/O speeds and latency of whatever storage solution you are using: especially if it's on network storage. Although compression is likely the biggest issue, upgrading the storage device can help too.

**Comment by OmegaShaman** [comment]
> Hello.
> 
> Just to confirm I am going through the exact same issue. I have the latest purpur version avaliable and the issue is still persisting. I have a 70 av playerbase and the chunks are fine only if I kick everyone else. As OP mentioned, the server itself is not lagging, TPS, CPU usage timings all look fine.

Had this issue for almost months now, is your map >350 GB? Mine is and I still can't figure out what's causing that lag.

**Comment by OmegaShaman** [comment]
> > Hello.
> > Just to confirm I am going through the exact same issue. I have the latest purpur version avaliable and the issue is still persisting. I have a 70 av playerbase and the chunks are fine only if I kick everyone else. As OP mentioned, the server itself is not lagging, TPS, CPU usage timings all look fine.
> 
> Had this issue for almost months now, is your map >350 GB? Mine is and I still can't figure out what's causing that lag.

I've also had that since August too, perhaps it may be due to JDK 16?

**Comment by Apprisco** [comment]
I ran into this issue on Paper itself, and although I reported it, even with a possible solution, nothing came to it.
From what I can tell, it's because Paper itself is extremely shitty at handling slower storage - even high-speed network storage latency was enough to make it shit itself with 30 players or so. Using a slower drive locally with lower latency was working much better. 

On the other hand, network compression can do the exact same thing, which is fun.
